src/main/java/graphql/execution/ExecutionStepInfoFactory.java

```diff
@@ -43,10 +43,9 @@ public ExecutionStepInfo newExecutionStepInfoForSubField(ExecutionContext execut
                 .arguments(argumentValues));
     }
 
-    public ExecutionStepInfo newExecutionStepInfoForListElement(ExecutionStepInfo executionInfo, int index) {
+    public ExecutionStepInfo newExecutionStepInfoForListElement(ExecutionStepInfo executionInfo, ResultPath indexedPath) {
         GraphQLList fieldType = (GraphQLList) executionInfo.getUnwrappedNonNullType();
         GraphQLOutputType typeInList = (GraphQLOutputType) fieldType.getWrappedType();
-        ResultPath indexedPath = executionInfo.getPath().segment(index);
```

Comment:
according to my memory profiler, this is dropping from 3-4% of help to just 1-2% (as expected)